File: JAVA/Shirt.java
class Shirt extends Clothing{   //class cucu yang mendapatkan turunan dari product dan clothing
    String color;   //atribut di class shirt
    String Sleeve_type;

    Shirt(){    //kontruktor kosong
    }

    Shirt(String color, String sleeve_type){    //konstruktor berparameter
        this.color = color;
        this.Sleeve_type = sleeve_type;
    }

    //get dan set atribut
    public String getColor() {
        return color;
    }

    public void setColor(String color) {
        this.color = color;
    }

    public String getSleeve_type() {
        return Sleeve_type;
    }

    public void setSleeve_type(String sleeve_type) {
        Sleeve_type = sleeve_type;
    }
}

File: CPP/PROGRAM/Shirt.cpp
#include <iostream>
#include <string>
#include "Clothing.cpp" //menginclude nama file orangtua untuk mewariskan object dari class orang tua
//using standard namespace
using namespace std;

//class shirt yang mewarisi dari kelas clothing
class Shirt : public Clothing{
    private:
        string color;
        string sleeve_type;

    public:
        Shirt(){
        }
        
        //getter dan setter
        void set_color(string color){
            this->color = color;
        }

        string get_color(){
            return this->color;
        }

        void set_sleeve_type(string sleeve_type){
            this->sleeve_type = sleeve_type;
        }

        string get_sleeve_type(){
            return this->sleeve_type;
        }

        ~Shirt(){
        }
};